### PR TITLE
chore: remove gorm in go structs where possible

### LINKED ISSUE #297: Optimize struct tags
Struct tag cleanup, validation, gorm, etc

### LINKED ISSUE #635: chore: remove gorm in schema

### PR DESCRIPTION
**Pull request by garrettladley**

### CODE REVIEW

File: backend/entities/models/comment.go [hunk 1]
```diff
@@ -5,16 +5,16 @@ import "github.com/google/uuid"
...
+	AskedBy   User      `gorm:"foreignKey:AskedByID;" json:"-" validate:"-"`
 
-	ClubID uuid.UUID `gorm:"type:uuid;not null" json:"-" validate:"uuid4"`
-	Club   Club      `gorm:"foreignKey:ClubID;not null" json:"-" validate:"-"`
+	ClubID uuid.UUID `json:"-" validate:"uuid4"`
+	Club   Club      `gorm:"foreignKey:ClubID;" json:"-" validate:"-"`
 
-	AnsweredByID *uuid.UUID `gorm:"type:uuid" json:"-" validate:"uuid4"`
+	AnsweredByID *uuid.UUID `json:"-" validate:"uuid4"`
 	AnsweredBy   *User      `gorm:"foreignKey:AnsweredBy" json:"-" validate:"-"`
```
Comment (DOOduneye) [inline backend/entities/models/comment.go:19]: why dont you remove gorm here?

File: backend/entities/models/contact.go [hunk 2]
```diff
@@ -43,10 +43,10 @@ func GetContentPrefix(contactType ContactType) string {
 type Contact struct {
 	Model
 
-	Type    ContactType `gorm:"type:varchar(255);uniqueIndex:idx_contact_type;not null" json:"type" validate:"required,max=255,oneof=facebook instagram x linkedin youtube github slack discord email customSite"`
-	Content string      `gorm:"type:varchar(255);not null" json:"content" validate:"required,max=255"`
+	Type    ContactType `json:"type" validate:"required,max=255,oneof=facebook instagram x linkedin youtube github slack discord email customSite"`
+	Content string      `json:"content" validate:"required,max=255"`
 
-	ClubID uuid.UUID `gorm:"foreignKey:ClubID;uniqueIndex:idx_contact_type;not null" json:"-" validate:"uuid4"`
+	ClubID uuid.UUID `gorm:"foreignKey:ClubID;" json:"-" validate:"uuid4"`
```
Comment (DOOduneye) [inline backend/entities/models/contact.go:49]: same

**Review by DOOduneye — CHANGES_REQUESTED** [review]
Why are some left and others gone?

**Comment by garrettladley** [comment]
> Why are some left and others gone?

encountered some oddities with gorm, state at review time was very much experimental. will see if i can remove more while not causing regressions

**Comment by garrettladley** [comment]
the reason i left in the `gorm:"many2many:` is because that was present in the docs, while other places the struct tags appeared to be optional. [see](https://gorm.io/docs/many_to_many.html)

**Comment by garrettladley** [comment]
closing thoughts, my observations where struct tags are necessary is as follows:

when there is a many to many ([see](https://gorm.io/docs/many_to_many.html))

when there is a has many, with a differently named foreign key ID

has many example from gorm
```go
// User has many CreditCards, UserID is the foreign key
type User struct {
  gorm.Model
  CreditCards []CreditCard
}

type CreditCard struct {
  gorm.Model
  Number string
  UserID uint
}
```

explicit has many foreign key required (forgive the contrived example)
```go
// User has many CreditCards, UserID is the foreign key
type User struct {
  gorm.Model
  CreditCards []CreditCard `gorm:"foreignKey:CustomerID;"` // required bc foreign key != <this struct name>ID
}

type CreditCard struct {
  gorm.Model
  Number string
  CustomerID uint
}
```

**Pull request merged**